### PR TITLE
fix: base64 decode bootstrap data

### PR DESCRIPTION
**Pull request by rudoi**
Signed-off-by: Andrew Rudoi <arudoi@newrelic.com>

**What this PR does / why we need it**:
Decodes bootstrap data. The bootstrap provider sets the bootstrap data as a base64-encoded string, so we must decode it.

Also snuck in a small typo fix. 

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/services/ec2/instances_test.go [hunk 1]
```diff
@@ -245,7 +245,8 @@ func TestCreateInstance(t *testing.T) {
 				},
 				Spec: clusterv1.MachineSpec{
 					Bootstrap: clusterv1.Bootstrap{
-						Data: pointer.StringPtr("user-data"),
+						// echo "user-data" | base64
+						Data: pointer.StringPtr("dXNlci1kYXRhCg=="),
```
Comment (vincepri) [inline pkg/cloud/services/ec2/instances_test.go]: Shouldn't the value in Bootstrap.Data be just a plain string?

File: pkg/cloud/services/ec2/instances.go [hunk 2]
```diff
@@ -204,7 +204,7 @@ func (s *Service) GetCoreSecurityGroups(scope *scope.MachineScope) ([]string, er
 	switch scope.Role() {
 	case "node":
 		// Just the common security groups above
-	case "controlplane":
+	case "control-plane":
```
Comment (vincepri) [inline pkg/cloud/services/ec2/instances.go]: Sidenote: We should probably document these roles in CAPI at some point

File: pkg/cloud/services/ec2/instances_test.go (hunk 1 above)
Comment (rudoi) [reply pkg/cloud/services/ec2/instances_test.go]: As in not base64? I dunno if a decision was made at some point to not do that but the bootstrapper's template func encodes it:

https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/c150edc4f00e11ba1cad905f18d4caff1ad5d1ae/cloudinit/utils.go#L31

**Comment by vincepri** [comment]
@rudoi Let's chat in today's meeting about the user-data and base64 encoding. I'm not sure if the kubeadm provider is encoding it, the original boostrap spec didn't account for it.

**Comment by detiber** [comment]
@vincepri I think base64 encoding makes sense, since for the general use case it avoids a lot of problems of embedding a yaml string in a yaml doc.

**Comment by vincepri** [comment]
/approve

Let's merge this in and fix it later if needed

**Pull request merged**